### PR TITLE
[mp] add github source

### PR DESCRIPTION
**Pull request by mattppal**
# Description
1. Add GitHub Source for data integration
2. Update instructions for testing 
3. Add documentation for adapting Source/Tap.

# How Has This Been Tested?

- [x] Methods described [here](https://github.com/mage-ai/mage-ai/tree/master/mage_integrations)

## Connection Test

image

## Stream Selection

image

## Full & Incremental Sync

Note: it does not appear that the tap supports running an incremental sync: https://github.com/MeltanoLabs/tap-github

image

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand area
- [x] I have made corresponding changes to the documentation

cc:

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Please test in the UI as well and provide screenshots for the testing results
* Test connection
* View and select streams
* Sync one stream to a destination (show logs of the source and show data in the destination table)
* If incremental sync is supported, please also test it

**Comment by mattppal** [comment]
- [x] Test connection
- [x] View and select streams
- [x] Sync one stream to a destination (show logs of the source and show data in the destination table)
- [x] If incremental sync is supported, please also test it
  - [ ] Does not appear to be supported 

Output: 

image

**Comment by mattppal** [comment]
@wangxiaoyou1993 

Something is broken with the URL values— they don't match what github is showing [in their docs](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28), e.g.

```json response

{
    "url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
    "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e",
    "node_id": "MDY6Q29tbWl0NmRjYjA5YjViNTc4NzVmMzM0ZjYxYWViZWQ2OTVlMmU0MTkzZGI1ZQ==",
    "html_url": "https://github.com/octocat/Hello-World/commit/6dcb09b5b57875f334f61aebed695e2e4193db5e",
    "comments_url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e/comments",
    "commit": {
      "url": "https://api.github.com/repos/octocat/Hello-World/git/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
      "author": {
        "name": "Monalisa Octocat",
        "email": "support@github.com",
        "date": "2011-04-14T16:00:49Z"
      },
      "committer": {
        "name": "Monalisa Octocat",
        "email": "support@github.com",
        "date": "2011-04-14T16:00:49Z"
      },
      "message": "Fix all the bugs",
      "tree": {
        "url": "https://api.github.com/repos/octocat/Hello-World/tree/6dcb09b5b57875f334f61aebed695e2e4193db5e",
        "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e"
      },
      "comment_count": 0,
      "verification": {
        "verified": false,
        "reason": "unsigned",
        "signature": null,
        "payload": null
      }
    },
    "author": {
      "login": "octocat",
      "id": 1,
      "node_id": "MDQ6VXNlcjE=",
      "avatar_url": "https://github.com/images/error/octocat_happy.gif",
      "gravatar_id": "",
      "url": "https://api.github.com/users/octocat",
      "html_url": "https://github.com/octocat",
      "followers_url": "https://api.github.com/users/octocat/followers",
      "following_url": "https://api.github.com/users/octocat/following{/other_user}",
      "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
      "organizations_url": "https://api.github.com/users/octocat/orgs",
      "repos_url": "https://api.github.com/users/octocat/repos",
      "events_url": "https://api.github.com/users/octocat/events{/privacy}",
      "received_events_url": "https://api.github.com/users/octocat/received_events",
      "type": "User",
      "site_admin": false
    },
    "committer": {
      "login": "octocat",
      "id": 1,
      "node_id": "MDQ6VXNlcjE=",
      "avatar_url": "https://github.com/images/error/octocat_happy.gif",
      "gravatar_id": "",
      "url": "https://api.github.com/users/octocat",
      "html_url": "https://github.com/octocat",
      "followers_url": "https://api.github.com/users/octocat/followers",
      "following_url": "https://api.github.com/users/octocat/following{/other_user}",
      "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
      "organizations_url": "https://api.github.com/users/octocat/orgs",
      "repos_url": "https://api.github.com/users/octocat/repos",
      "events_url": "https://api.github.com/users/octocat/events{/privacy}",
      "received_events_url": "https://api.github.com/users/octocat/received_events",
      "type": "User",
      "site_admin": false
    },
    "parents": [
      {
        "url": "https://api.github.com/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
        "sha": "6dcb09b5b57875f334f61aebed695e2e4193db5e"
      }
    ]
  }
]
```

**Comment by mattppal** [comment]
This appears to only be an issue when syncing the tap via the mage ui, the command line doesn't have this problem 
image

**Comment by mattppal** [comment]
image

image

@wangxiaoyou1993 Still seeing the same issue with removed paths in the destination (but not in the raw data)

File: mage_integrations/README.md [hunk 1]
```diff
@@ -225,6 +230,8 @@ python3 mage_integrations/sources/[INTEGRATION]/__init__.py \
 
 ### Perform a sample query
 
+### Perform a sample query
```
Comment (wangxiaoyou1993) [inline mage_integrations/README.md:233]: nit: remove duplicate title

File: mage_integrations/README.md [hunk 2]
```diff
@@ -142,6 +142,11 @@ python3 mage_integrations/sources/[INTEGRATION]/__init__.py \
   --state mage_integrations/TEST_STATE.json > mage_integrations/TEST_OUTPUT
 ```
 
+python3 mage_integrations/sources/github/__init__.py \
```
Comment (wangxiaoyou1993) [inline mage_integrations/README.md:145]: nit: remove duplicate command

**Comment by mattppal** [comment]
@wangxiaoyou1993 fixed your comments, added note about incremental sync in the readme, and confirmed data is accurate.

image

File: mage_integrations/mage_integrations/sources/github/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  - Stargazers
+  - Teams
+  - Team Members
+  - Team Memberships
+- Outputs the schema for each resource
+
+
+Note: at this time, incremental sync is not supported for any endpoint. In order to support, we'll need to modify the `./tap_github/streams.py` `IncrementalStream` class to support incremental sync. This appears to be due to a malformed bookmark value
+
+```json
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/github/README.md:57]: you don't need to show the state message here. users won't need to deal with it.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**